src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -18,64 +21,75 @@
-                                                      @RequestParam("thumbnailImage") MultipartFile thumbnailImage,
-                                                      @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse patchProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
 
         ProductEntity patchedProduct = productService.patchProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(patchedProduct);
+        return productMapper.toProductResponse(patchedProduct);
     }
 
     @PutMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<ProductEntity> updateProduct(@PathVariable Long productId,
-                                                       @ModelAttribute ProductUpdateRequest request,
-                                                       @RequestParam("thumbnailImage") MultipartFile thumbnailImage,
-                                                       @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse updateProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
+
         ProductEntity updatedProduct = productService.updateProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(updatedProduct);
+        return productMapper.toProductResponse(updatedProduct);
     }
 
     @DeleteMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
     public ResponseEntity<Void> deleteProduct(@PathVariable Long productId) throws IOException {
+
         productService.deleteProduct(productId);
+
         return ResponseEntity.noContent().build();
     }
 
     @DeleteMapping("/delete-all")
     @PreAuthorize("hasRole('SELLER')")
     public ResponseEntity<Void> deleteAllProductsForCurrentSeller() throws IOException {
+
         productService.deleteAllProductsForCurrentSeller();
+
         return ResponseEntity.noContent().build();
     }
 
     @GetMapping
     public List<ProductEntity> getAllProducts(
             @RequestParam(defaultValue = "0") int offset,
             @RequestParam(defaultValue = "4") int limit) {
+
         return productService.getProducts(offset, limit);
```

Comment:
please try the below experiment here.
1. put debugging break point here and run app in debug mode
2. open evaluation tool (ALF+F8) , then put the below code in it and evaluate:
```java
CategoryEntity c = CategoryEntity.builder()
        .id(1L)
        .name("blah")
        .build();

ProductEntity pe = ProductEntity.builder()
        .id(1L)
        .name("blah")
        .category(c)
        .build();

productMapper.toProductResponse(pe);
```
note, what is category in your ProductResponse ?
compaire the same with category field in CategoryResponse - how is that mapped ?

long story short: we want to return plain response here, no complex data structures :)

Reply:
So you haven't seen the return with the seller yet), but what data do we need to return? For now, we can choose, because the object is returned.
```
https://drive.google.com/file/d/1njKp383soxPctm-MJ5cBLp5kIcCzWyUL/view?usp=drive_link
```

Reply:
hey, please ignore this.
you've answered above about this: https://github.com/Hutiir-Craft-Food/FoodBackend/pull/35#discussion_r1796889513 
it makes sense now. 
initially i was implying to changing ProductResponse, so that instead of `category: object` it had `categoryId: number`
but it seems your point of returning full category as object is fair.